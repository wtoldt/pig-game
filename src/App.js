import React, { Component } from 'react';
import Board from './components/Board/Board';
import PlayerPanel from './components/PlayerPanel/PlayerPanel';
import { PLAYER_STATE } from './App.consts';

import './App.scss';

export default class App extends Component {

  state = {
    p1: {name: 'P1', ...PLAYER_STATE},
    p2: {name: 'P2', ...PLAYER_STATE},
    currentPlayer: 'P1',
    winner: undefined
  };

  render() {
    const { p1, p2, currentPlayer } = this.state;
    return (
      <div className="pg-app">
        <Board>
          <PlayerPanel player={p1} active={currentPlayer === p1.name} left />
          <PlayerPanel player={p2} active={currentPlayer === p2.name} right />
        </Board>
      </div>
    );
  }
}
