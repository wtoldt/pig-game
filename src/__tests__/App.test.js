import { render, screen } from '@testing-library/react';
import App from '../App';

const defaultProps = {};

function getWrapper(props={}) {
  return render(
    <App {...defaultProps} {...props} />
  )
}

describe('App Component', () => {

  describe('Snapshot testing', () => {
    it('Default snapshot', () => {
      const wrapper = getWrapper();
      expect(wrapper).toMatchSnapshot();
    });

    it('Has text pig game', () => {
      //assemble
      const wrapper = getWrapper();
      expect(screen.getByText(/pig game/i)).toBeInTheDocument();
    });
  });
});
