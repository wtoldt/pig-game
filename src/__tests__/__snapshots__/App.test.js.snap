// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Snapshot testing Default snapshot 1`] = `
<div>
  <div
    class="pg-app"
  >
    <div
      class="pg-board"
    >
      <div
        class="pg-player-panel active left"
      >
        <div
          class="player-name-container"
        >
          <div
            class="player-name"
          >
            P1
          </div>
          <div
            class="overall-score"
          >
            0
          </div>
        </div>
        <div
          class="current-score-container"
        >
          <div
            class="current-score"
          >
            0
          </div>
        </div>
      </div>
      <div
        class="pg-game-controls"
      >
        <div
          class="reset-btn-container"
        >
          <button>
            Reset Game
          </button>
        </div>
        <div
          class="die-container"
        >
          <div
            class="pg-die"
          >
            <div
              class="row"
            >
              <div
                class="cell"
                id="c1"
              >
                 
              </div>
              <div
                class="cell"
                id="c2"
              >
                 
              </div>
              <div
                class="cell"
                id="c3"
              >
                 
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="cell"
                id="c4"
              >
                 
              </div>
              <div
                class="cell"
                id="c5"
              >
                üêΩ
              </div>
              <div
                class="cell"
                id="c6"
              >
                 
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="cell"
                id="c7"
              >
                 
              </div>
              <div
                class="cell"
                id="c8"
              >
                 
              </div>
              <div
                class="cell"
                id="c9"
              >
                 
              </div>
            </div>
          </div>
        </div>
        <div
          class="roll-btn-container"
        >
          <button>
            Roll
          </button>
        </div>
        <div
          class="hold-btn-container"
        >
          <button>
            Hold
          </button>
        </div>
      </div>
      <div
        class="pg-player-panel right"
      >
        <div
          class="player-name-container"
        >
          <div
            class="player-name"
          >
            P2
          </div>
          <div
            class="overall-score"
          >
            0
          </div>
        </div>
        <div
          class="current-score-container"
        >
          <div
            class="current-score"
          >
            0
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
